src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -160,13 +165,27 @@ public override void Initialize()
                 {
                     _openedEventHandler(new ClusterOpenedEvent(ClusterId, Settings, stopwatch.Elapsed));
                 }
+
+                MonitorServersAsync().ConfigureAwait(false);
+
+                if (Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv)
+                {
+                    var dnsEndPoint = (DnsEndPoint)Settings.EndPoints.Single();
+                    var lookupDomainName = dnsEndPoint.Host;
+                    var dnsMonitor = new DnsMonitor(this, lookupDomainName, _monitorServersCancellationTokenSource.Token);
+                    var thread = new Thread(dnsMonitor.Start);
+                    thread.Start();
```

Comment:
This is where we start the new DnsMonitor thread IFF it is needed.

Reply:
Please let me know about this one.

Reply:
Also, I suspect that you'll want to start this thread in DnsMonitor itself once you interfacify DnsMonitor.  It will make it easier to mock.